Decode/PerfItemValue.cs

```diff
@@ -2234,12 +2121,44 @@ private void Value128Append(StringBuilder sb, int elementIndex)
                 case EventHeaderFieldFormat.Uuid:
                     PerfConvert.GuidAppend(sb, this.GetGuid(elementIndex));
                     break;
-                case EventHeaderFieldFormat.IPv6:
+                case EventHeaderFieldFormat.IPAddress:
+                case EventHeaderFieldFormat.IPAddressObsolete:
                     PerfConvert.IPv6Append(sb, this.GetIPv6(elementIndex));
                     break;
             }
         }
 
+        private void StringAppend(StringBuilder sb, PerfConvertOptions convertOptions, Text.Encoding defaultEncoding)
```

Comment:
Note: Factored-out a subroutine to reduce the number of gotos and code duplication. Otherwise the new formatting semantics would have required 2 more copies of this logic.